sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/MapState.java

```diff
@@ -30,64 +30,50 @@
 public interface MapState<K, V> extends State {
 
   /**
-   * Returns the value to which the specified key is mapped in the state.
-   */
-  V get(K key);
-
-  /**
    * Associates the specified value with the specified key in this state.
    */
   void put(K key, V value);
 
   /**
-   * If the specified key is not already associated with a value (or is mapped
-   * to {@code null}) associates it with the given value and returns
-   * {@code null}, else returns the current value.
+   * A deferred read-followed-by-write.
+   *
+   * <p>When {@code read()} is called on the result, it forces a read of the map and reconciliation
+   * with any pending modifications.
+   *
+   * <p>If the specified key is not already associated with a value (or is mapped to {@code null})
+   * associates it with the given value and returns {@code null}, else returns the current value.
    */
-  V putIfAbsent(K key, V value);
+  ReadableState<V> putIfAbsent(K key, V value);
```

Comment:
I added this method because of Map of JDK1.8. It can't reduce access to BackendState and we don't have the problem of concurrency, it may be redundant.